src/oidcc_profile.erl

```diff
@@ -0,0 +1,152 @@
+%%%-------------------------------------------------------------------
+%% @doc OpenID Profile Utilities
+%% @end
+%% @since 3.2.0
+%%%-------------------------------------------------------------------
+-module(oidcc_profile).
+
+-feature(maybe_expr, enable).
+
+-include("oidcc_client_context.hrl").
+-include("oidcc_provider_configuration.hrl").
+
+-export([apply_profiles/2]).
+
+%% @private
+-spec apply_profiles(ClientContext, Opts) ->
+    {ok, ClientContext, Opts} | {error, oidcc_client_context:error()}
+when
+    ClientContext :: oidcc_client_context:t(),
+    Opts :: map().
```

Comment:
I believe we should define a type for the opts map and a specific type for the profile atoms.

Reply:
Should that live in `oidcc_client_context` or `oidcc_profile`?

Reply:
We‘ve located the types close to the code so far. (Which resulted in public modules where only the types are public but all fns are private.)

I think we should stick with that for now. (=> `oidcc_profile`)